### PR TITLE
Use background color for input toggle active state

### LINKED ISSUE #36439: Toggle button focus is difficult to determine
- VSCode Version: Code 1.17.2 (b813d12980308015bcd2b3a2f6efa5c810c33ba5, 2017-10-16T13:57:00.652Z)
- OS Version: Linux x64 4.10.0-35-generic
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
terminal-tabs|Tyr|0.1.1

---

Active + not focused:

image

Active + focused:

image

### PR DESCRIPTION
**Pull request by miguelsolorio**
Fixes #36439 as discussed in https://github.com/microsoft/vscode/issues/36439#issuecomment-428749511

## Dark
image

## Light
image

This adds `inputOption.activeBackground` on active toggle states (search, find widget, etc.). This also sets `inputOption.activeBorder` to transparent as will only use outline on toggles for focus, as discussed in https://github.com/microsoft/vscode/issues/36439#issuecomment-428749511. The new color is re-using `focusBorder` so it works well with other themes. Sample below:

ezgif com-gif-maker(47)

cc @Tyriar @sandy081 @roblourens @bpasero @aeschli

### CODE REVIEW

File: src/vs/platform/theme/common/colorRegistry.ts [hunk 1]
```diff
@@ -203,7 +203,8 @@ export const widgetShadow = registerColor('widget.shadow', { dark: '#000000', li
 export const inputBackground = registerColor('input.background', { dark: '#3C3C3C', light: Color.white, hc: Color.black }, nls.localize('inputBoxBackground', "Input box background."));
 export const inputForeground = registerColor('input.foreground', { dark: foreground, light: foreground, hc: foreground }, nls.localize('inputBoxForeground', "Input box foreground."));
 export const inputBorder = registerColor('input.border', { dark: null, light: null, hc: contrastBorder }, nls.localize('inputBoxBorder', "Input box border."));
-export const inputActiveOptionBorder = registerColor('inputOption.activeBorder', { dark: '#007ACC', light: '#007ACC', hc: activeContrastBorder }, nls.localize('inputBoxActiveOptionBorder', "Border color of activated options in input fields."));
+export const inputActiveOptionBorder = registerColor('inputOption.activeBorder', { dark: '#007ACC00', light: '#007ACC00', hc: '#007ACC00' }, nls.localize('inputBoxActiveOptionBorder', "Border color of activated options in input fields."));
```
Comment (misolori) [inline src/vs/platform/theme/common/colorRegistry.ts]: @isidorn Also, not sure if you can help me understand why setting `inputOption.activeBorder` to `null` results in the colors not updating when being toggled via `workbench.colorCustomizations`. Ideally this should be `null` but can only get it to work when setting it to a transparent color.

**Comment by misolori** [comment]
Update the colors for high contrast mode:

**Review by roblourens — APPROVED** [review]
Looks great! Does this affect other checkboxes or just the ones in the find input? Does it look good with other popular themes?

**Comment by misolori** [comment]
@roblourens doesn't affect other checkboxes, just the toggle ones like exclude and search widget. 

Here's what it looks like in other themes, some explicitly set `inputOption.activeBorder` so it doesn't look like much of a change for them:

gif

**Comment by roblourens** [comment]
Ok cool, how about the checkboxes in the settings editor? (sorry I could build it myself but my repo is in the middle of something else)

**Comment by misolori** [comment]
@roblourens doesn't affect checkboxes in the settings editor as it's not a toggle:

image

**Comment by isidorn** [comment]
@misolori  I was out on vacation, sorry for not responding.
This looks good, I will provide more feedback once I selfhost on it for a couple of days.

**Pull request merged**